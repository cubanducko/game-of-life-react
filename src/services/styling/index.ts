export * from "./spacing";
export * from "./breakpoints";
export * from "./theme";
export * from "./GlobalStyles.component";
