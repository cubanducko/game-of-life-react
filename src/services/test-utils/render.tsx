import {
  render as rtlRender,
  RenderOptions,
  RenderResult,
} from "@testing-library/react";
import { FunctionComponent, PropsWithChildren, ReactElement } from "react";
import { createTheme, Theme } from "services/styling";
import { ThemeProvider } from "styled-components";

const wrapper = ({
  theme: themeVariables,
}: IWrapperOptions): FunctionComponent => {
  const RTLWrapper = ({
    children,
  }: PropsWithChildren<unknown>): ReactElement => {
    const theme = createTheme(themeVariables);

    return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
  };

  return RTLWrapper;
};

export interface IWrapperOptions {
  theme?: Theme;
}

const renderWithProviders = (
  element: ReactElement,
  options: IWrapperOptions = {}
): RenderResult => {
  return rtlRender(element, {
    wrapper: wrapper(options),
  } as RenderOptions);
};

export { default as userEvent } from "@testing-library/user-event";
export * from "@testing-library/react";
export { renderWithProviders as render };
