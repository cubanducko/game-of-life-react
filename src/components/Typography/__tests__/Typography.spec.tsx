import { render } from "services/test-utils";
import { H1 } from "../H1.component";
import { H2 } from "../H2.component";

describe("Typography", () => {
  it("renders H1 correctly", () => {
    const { container } = render(<H1>Example title</H1>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders H2 correctly", () => {
    const { container } = render(<H2>Example subtitle</H2>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
