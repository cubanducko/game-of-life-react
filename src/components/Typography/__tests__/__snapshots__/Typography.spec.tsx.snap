// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography renders H1 correctly 1`] = `
.c0 {
  font-size: 64px;
  line-height: 80px;
  font-weight: 700;
  margin: 0;
}

<h1
  class="c0"
>
  Example title
</h1>
`;

exports[`Typography renders H2 correctly 1`] = `
.c0 {
  font-size: 24px;
  line-height: 32px;
  font-weight: 400;
  margin: 0;
}

<h2
  class="c0"
>
  Example subtitle
</h2>
`;
